### PR TITLE
Add examples for .$type method and text-based enum columns

### PR DESCRIPTION
**Pull request by meech-ward**
These updates are from the 0.23.0 release, it's all in that changelog. 
<https://github.com/drizzle-team/drizzle-orm/releases/tag/0.23.0>
Most of the documentation added here already exists in the older documentation README.md files.

### CODE REVIEW

**Comment by meech-ward** [comment]
includes changes from https://github.com/drizzle-team/drizzle-orm-docs/pull/33

File: pages/docs/column-types/mysql.mdx [hunk 1]
```diff
@@ -538,6 +557,17 @@ CREATE TABLE `table` (
 ```
 </Section>
 
+### Customizing column data type
+
+Every column builder has a `.$type()` method, which allows you to customize the data type of the column. This is useful, for example, with branded types.
+
+```ts
+const users = mysqlTable('users', {
+  id: serial('id').$type<UserId>().primaryKey(),
```
Comment (AlexBlokh) [inline pages/docs/column-types/mysql.mdx:566]: I suppose we need to declare branded type above and then show example usage below

File: pages/docs/column-types/pg.mdx [hunk 2]
```diff
@@ -593,6 +593,17 @@ CREATE TABLE IF NOT EXISTS "table" (
 ```
 </Section>
 
+### Customizing column data type
+
+Every column builder has a `.$type()` method, which allows you to customize the data type of the column. This is useful, for example, with branded types.
+
+```ts
+const users = pgTable('users', {
+  id: serial('id').$type<UserId>().primaryKey(),
```
Comment (AlexBlokh) [inline pages/docs/column-types/pg.mdx:602]: Same here

File: pages/docs/column-types/sqlite.mdx [hunk 3]
```diff
@@ -118,8 +118,32 @@ CREATE TABLE `table` (
...
+
 </Section>
 
+### Customizing column data type
+
+Every column builder has a `.$type()` method, which allows you to customize the data type of the column. This is useful, for example, with branded types.
+
+```ts
+const users = sqliteTable('users', {
+  id: integer('id').$type<UserId>().primaryKey(),
```
Comment (AlexBlokh) [inline pages/docs/column-types/sqlite.mdx:142]: Same here

**Comment by AlexBlokh** [comment]
@meech-ward I've faced conflict problems and just migrated all the changes manually, sorry for a massive delay

thank you so much for your contribution!

**Pull request closed without merging**